.github/workflows/stylecheck.yml

```diff
@@ -0,0 +1,39 @@
+name: Style check
+
+on:
+  pull_request:
+    branches:
+      - master
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  style-check:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v2
```

Comment:
`checkout@v2` is too old.  We should bump it to `v4`.  Since this PR is doing refactoring, it is a good chance to do a global search/replacement for GitHub actions like this.

Reply:
I changed all the `checkout` uses to `v4`.